src/lib/components/grid/types.ts

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+
+export type ResponsiveValue<T> = T | {
+  default?: T; // fallback value that applies when no breakpoint matches
+  sm?: T;      // applies from 0px up
+  md?: T;      // > 500px container width
+  lg?: T;      // > 800px container width
+};
+
+export const GRID_ALIGN_ITEMS = ["start", "end", "center", "stretch", "baseline"] as const;
+export type GridAlignItems = (typeof GRID_ALIGN_ITEMS)[number];
+
+export const GRID_JUSTIFY_ITEMS = ["start", "end", "center", "stretch"] as const;
+export type GridJustifyItems = (typeof GRID_JUSTIFY_ITEMS)[number];
+
+export const GRID_ALIGN_CONTENT = [
+  "start",
+  "end",
+  "center",
+  "stretch",
+  "space-around",
+  "space-between",
+  "space-evenly"
+] as const;
+export type GridAlignContent = (typeof GRID_ALIGN_CONTENT)[number];
+
+export const GRID_JUSTIFY_CONTENT = [
+  "start",
+  "end",
+  "center",
+  "stretch",
+  "space-around",
+  "space-between",
+  "space-evenly"
+] as const;
+export type GridJustifyContent = (typeof GRID_JUSTIFY_CONTENT)[number];
+
+export const GRID_ALIGN_SELF = ["start", "end", "center", "stretch", "baseline"] as const;
+export type GridAlignSelf = (typeof GRID_ALIGN_SELF)[number];
+
+export const GRID_JUSTIFY_SELF = ["start", "end", "center", "stretch"] as const;
+export type GridJustifySelf = (typeof GRID_JUSTIFY_SELF)[number];
+
+export type GridSpan = number | "auto";
+export type GridLine = number | "auto" | `span ${number}`;
+
+export type SimpleGridColumns = number | Record<string, number>;
+
+export interface GridItemProps {
```

Comment:
Typically we colocate props definitions with the component that uses them. Can we do that here too?

And a very small nit: we typically use `type` over `interface`.